parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh

```diff
@@ -25,9 +25,41 @@ installDeps() {
     fi
 }
 
+addMarinerNvidiaRepo() {
+    MARINER_NVIDIA_REPO_FILEPATH="/etc/yum.repos.d/mariner-nvidia.repo"
+    touch "${MARINER_NVIDIA_REPO_FILEPATH}"
+    cat << EOF > "${MARINER_NVIDIA_REPO_FILEPATH}"
+[mariner-official-nvidia]
+name=CBL-Mariner Official Nvidia 2.0 x86_64
+baseurl=https://packages.microsoft.com/cbl-mariner/2.0/prod/nvidia/x86_64
+gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY
+gpgcheck=1
+repo_gpgcheck=1
+enabled=1
+skip_if_unavailable=True
+sslverify=1
+EOF
+    set -x
+}
+
 downloadGPUDrivers() {
-    echo "GPU drivers not yet supported for Mariner"
-    exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
+    if ! dnf_install 30 1 600 cuda; then
+      exit $ERR_APT_INSTALL_TIMEOUT
+    fi
+}
+
+installNvidiaContainerRuntime() {
+    for nvidia_package in nvidia-container-runtime nvidia-container-toolkit libnvidia-container-tools libnvidia-container1; do
```

Comment:
this looks to be installing 1.9/3.9. this package set is correct for that version, but 3.11/1.11 added nvidia-container-toolkit-base and 1.10+ is a prereq for cgroupv2 which we should explore for mariner. 

can you possibly bump this to 3.11 to match?

```
  libnvidia-container-tools-1.9.0-3.cm2.x86_64                                  
  libnvidia-container1-1.9.0-3.cm2.x86_64                                       
  nvidia-container-runtime-3.9.0-1.cm2.x86_64                                   
  nvidia-container-toolkit-1.9.0-3.cm2.x86_64   
```

Reply:
tangentially, from logs looks like this takes 60 sec on vanilla marketplace image which isn't too shabby. would be nice if we can pre-cache those via the equivalent of `apt-get install --download-only` or something to compare the times. not super critical

Reply:
Sure, I can bump the package versions to 1.11/3.11

Reply:
thanks! I'd also suggest pinning this in the script so we can control upgrades and not pull latest from mariner distro. 

unless you're planning to control it that way deliberately, but that would be a little funky imo

Reply:
I have opened a PR to bump the related packages: https://github.com/microsoft/CBL-Mariner/pull/3813

Reply:
Have added version enforcement when installing these packages